instructure/canvas-ios
# Bulk publish logic

refs: MBL-17300
affects: Teacher
release note: none

test plan:
- Turn on feature flag.
- Start a bulk publish.
- Check if progress updates on the progress screen.
- Wait for it to complete.
- Go to modules screen, check if publish states are correctly updated or not.
- Repeat the publish but this time dismiss the progress screen and wait on the modules list.
- Progress indicators should disappear and published states should update.
- Repeat the publish but cancel the operation.

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td><img src="https://github.com/instructure/canvas-ios/assets/72396990/9a6af6fd-e382-48ef-b379-30d245ca6880" maxHeight=500></td>
<td><img src="https://github.com/instructure/canvas-ios/assets/72396990/3dbd78f0-3443-418d-a07f-fb45adabb495" maxHeight=500></td>
</tr>
<tr>
<td><img src="https://github.com/instructure/canvas-ios/assets/72396990/d5439ea5-d885-4ae5-b5af-7cadcbbfca74" maxHeight=500></td>
</tr>
</table>

## Checklist
